src/virtualenv/activation/via_template.py

```diff
@@ -41,6 +41,10 @@ def _generate(self, replacements, templates, to_folder, creator):
         for template in templates:
             text = self.instantiate_template(replacements, template, creator)
             dest = to_folder / self.as_name(template)
+            # remove the file if it already exists - this prevents permission
+            # errors when the dest is not writeable
```

Comment:
we can only remove if the parent directory is write-able, no? why would this file be not write-able and the parent be so?